x/stablestake/keeper/msg_server_unbond.go

```diff
@@ -9,12 +9,15 @@ import (
 
 func (k msgServer) Unbond(goCtx context.Context, msg *types.MsgUnbond) (*types.MsgUnbondResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
+	pool, found := k.GetPool(ctx, msg.PoolId)
+	if !found {
+		return nil, types.ErrPoolNotFound
+	}
 
-	params := k.GetParams(ctx)
 	creator := sdk.MustAccAddressFromBech32(msg.Creator)
-	redemptionRate := k.GetRedemptionRate(ctx)
+	redemptionRate := k.GetRedemptionRateForPool(ctx, pool)
```

Comment:
I think this method is better as we are calculating it, then no need to keep redemption rate member in pool